drivers/adc/adc_stm32.c

```diff
@@ -248,6 +248,18 @@ static int check_buffer_size(const struct adc_sequence *sequence,
 	return 0;
 }
 
+static uint32_t adc_stm32_getChannelByIndx(const uint8_t index) {
+	uint32_t channel = __LL_ADC_DECIMAL_NB_TO_CHANNEL(index);
+#if DT_N_S_soc_S_adc_40012000_P_ch17_internal == 1
```

Comment:
You know this part is soc specific, right ?

Reply:
I only know the stm32f412 MCU so I don't know how to make my code fitting to others st archs.

In general this code is a demo to show, how to add the internal channels to the stm32f412 by using the dts and some minor code patch of the stm32 adc driver.

I am happy to get some advice and execute the changes to make this commit better ;)